src/llama_stack_client/lib/agents/agent.py

```diff
@@ -96,16 +97,12 @@ def get_agent_config(
             agent_config["tool_config"] = tool_config
         if tools is not None:
             toolgroups: List[Toolgroup] = []
-            client_tools: List[ClientTool] = []
-
             for tool in tools:
                 if isinstance(tool, str) or isinstance(tool, dict):
                     toolgroups.append(tool)
-                else:
-                    client_tools.append(tool)
 
             agent_config["toolgroups"] = toolgroups
-            agent_config["client_tools"] = [tool.get_tool_definition() for tool in client_tools]
+            agent_config["client_tools"] = [tool.get_tool_definition() for tool in AgentUtils.get_client_tools(tools)]
```

Comment:
I think the same thing also needs to be added for AsyncAgent, sorry I need to remove this duplicate code.